consul/src/main/java/com/linecorp/armeria/client/consul/ConsulEndpointGroupBuilder.java

```diff
@@ -0,0 +1,129 @@
+
+    @Nullable
+    private URI consulUri;
+    @Nullable
+    private ConsulClient consulClient;
+    @Nullable
+    private String token;
+
+    private boolean useHealthyEndpoints;
+
+    ConsulEndpointGroupBuilder(String serviceName) {
+        this.serviceName = requireNonNull(serviceName, "serviceName");
+    }
+
+    /**
+     * Sets the specified {@code consulUri}.
+     */
+    public ConsulEndpointGroupBuilder consulUri(URI consulUri) {
+        this.consulUri = requireNonNull(consulUri, "consulUri");
+        return this;
+    }
+
+    /**
+     * Sets the specified {@code consulUri}.
+     */
+    public ConsulEndpointGroupBuilder consulUri(String consulUri) {
+        requireNonNull(consulUri, "consulUri");
+        checkArgument(!consulUri.isEmpty(), "consulUri can't be empty");
+        this.consulUri = URI.create(consulUri);
+        return this;
+    }
+
+    /**
+     * Sets the interval between fetching registry requests.
+     * If not set, {@value #DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS} is used by default.
+     */
+    public ConsulEndpointGroupBuilder registryFetchInterval(Duration registryFetchInterval) {
+        requireNonNull(registryFetchInterval, "registryFetchInterval");
+        final long seconds = registryFetchInterval.getSeconds();
+        checkArgument(seconds > 0, "registryFetchInterval.getSeconds(): %s (expected: > 0)", seconds);
+        return registryFetchIntervalSeconds(seconds);
+    }
+
+    /**
+     * Sets the interval between fetching registry requests.he interval between fetching registry requests.
+     * If not set {@value #DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS} is used by default.
+     */
+    public ConsulEndpointGroupBuilder registryFetchIntervalSeconds(long registryFetchIntervalSeconds) {
+        checkArgument(registryFetchIntervalSeconds > 0, "registryFetchIntervalSeconds: %s (expected: > 0)",
+                      registryFetchIntervalSeconds);
+        this.registryFetchIntervalSeconds = registryFetchIntervalSeconds;
+        return this;
+    }
+
+    /**
+     * Sets the specified {@code token} for accessing Consul server.
+     */
+    public ConsulEndpointGroupBuilder token(String token) {
+        this.token = requireNonNull(token, "token");
+        return this;
+    }
+
+    /**
+     * Sets whether to use <a href="https://www.consul.io/api/health.html">Health HTTP endpoint</a>.
+     * Before enabling this feature, make sure that your target endpoints are health-checked by Consul.
+     *
+     * @see ConsulUpdatingListenerBuilder#checkUri(URI)
+     */
+    public ConsulEndpointGroupBuilder useHealthEndpoints(boolean useHealthyEndpoints) {
+        this.useHealthyEndpoints = useHealthyEndpoints;
+        return this;
+    }
+
+    /**
+     * Returns a newly-created {@link ConsulEndpointGroup}.
+     */
+    public ConsulEndpointGroup build() {
+        if (consulClient == null) {
+            consulClient = new ConsulClient(consulUri, token);
```

Comment:
How about checking whether `consulUri` is `null` and throwing an exception if it is?
Or we can remove the `ConsulClient` setter which is only used in test and accept the uri in the constructor so that `consulUri` is not null.

Reply:
I refactored those codes.